api/client/cli.go

```diff
@@ -160,6 +162,13 @@ func NewDockerCli(in io.ReadCloser, out, err io.Writer, clientFlags *cli.ClientF
 		}
 		cli.configFile = configFile
 
+		// Set the CLI's API version
+		if ver := os.Getenv("DOCKER_API_VERSION"); ver != "" {
+			api.Version = version.Version(ver)
+		} else if configFile.APIVersion != "" {
+			api.Version = version.Version(configFile.APIVersion)
```

Comment:
-1 for all 3 bullets. This is a power user feature and its entire point is to unrestict things. For instance as long as feature is there a Docker client 1.8 can even call into engine 1.0. This is the purpose of this pr.